### PR TITLE
WIP: Issue 151 Include :_module-type: variable into the template

### PR DESCRIPTION
**Pull request by emmurphy1**
[ Issue 151 Include :_module-type: variable into the template](https://github.com/redhat-documentation/modular-docs/issues/151)

### CODE REVIEW

**Review by rolfedh — APPROVED** [review]

**Review by fbolton — APPROVED** [review]
LGTM

**Comment by mikemckiernan** [comment]
The change looks good, so formally, that is my approval.

That said:
- The module type is not customer-facing, or at least not compelling to make customer-facing.  Ideally, we would not add this attribute to the docs.
- If the change *must* be made, I'd prefer that we amend the code to accept lowercase for the type.

I have the second bullet working in a playpen, but maven is misbehaving _for me_, so I have not submitted a PR. The gist is to change the foll line from "attValue.toString()" to "attValue.toString.toUpperCase()".
https://github.com/redhataccess/pantheon/blob/master/pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/MetadataExtractorTreeProcessor.java#L150
Do we want to hold off for a code change?

File: modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc [hunk 1]
```diff
@@ -5,6 +5,13 @@ Base the file name and the ID on the module title. For example:
 * Title: = My concept module A
 ////
 
+////
+Indicate the module type in one of the following
+ways:
+Add the prefix con- or con_ to the file name.
+Add `:_module-type: CONCEPT` before the module ID.
```
Comment (rkratky) [inline modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc:12]: This being a comment (i.e. it won't be rendered anywhere), I don't see why the code should be enclosed in backticks.

```suggestion
Add the following attribute before the module ID:
:_module-type: CONCEPT
```

File: modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc [hunk 2]
```diff
@@ -7,6 +7,14 @@ Base the file name and the ID on the module title. For example:
 The ID is an anchor that links to the module. Avoid changing it after the module has been published to ensure existing links are not broken.
 ////
 
+////
+Indicate the module type in one of the following
+ways:
+Add the prefix proc- or proc_ to the file name.
+Add `:_module-type: PROCEDURE` before the module ID.
```
Comment (rkratky) [inline modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc:14]:
```suggestion
Add the following attribute before the module ID:
:_module-type: PROCEDURE
```

File: modular-docs-manual/files/TEMPLATE_REFERENCE_reference-material.adoc [hunk 3]
```diff
@@ -7,6 +7,13 @@ Base the file name and the ID on the module title. For example:
 The ID is an anchor that links to the module. Avoid changing it after the module has been published to ensure existing links are not broken.
 ////
 
+////
+Indicate the module type in one of the following
+ways:
+Add the prefix ref- or ref_ to the file name.
+Add `:_module-type: REFERENCE` before the module ID.
```
Comment (rkratky) [inline modular-docs-manual/files/TEMPLATE_REFERENCE_reference-material.adoc:14]:
```suggestion
Add the following attribute before the module ID:
:_module-type: REFERENCE
```

**Review by rkratky — CHANGES_REQUESTED** [review]

**Comment by rkratky** [comment]
@mikemckiernan,

> formally, that is my approval

I suggest you use the GH "Review" functionality to record your approval.

> I'd prefer that we amend the code to accept lowercase for the type

Agree, but that's a Pv2 matter (and as such, it should have no bearing on what happens in this repo).

**Review by yzimmerm — APPROVED** [review]

**Review by mikemckiernan — APPROVED** [review]
/lgtm

Maven behaved for me yesterday, so I submitted a PR to accept lowercase.  Dunno if anyone else cares or cares to wait.

https://github.com/redhataccess/pantheon/pull/562

**Review by sterobin — APPROVED** [review]

**Review by Preeticp — APPROVED** [review]
LGTM

**Pull request merged**